src/WebpackTools/middlewares/__tests__/DevProxy.spec.js

```diff
@@ -1,34 +1,146 @@
-jest.mock('http-proxy-middleware');
+let devProxy;
+const request = require('supertest');
+const nock = require('nock');
+const express = require('express');
 
-const proxyMiddleware = require('http-proxy-middleware');
+const TARGET = 'https://proxytarget.test';
+const TARGET_UNSECURE = 'http://proxytarget.test';
+const logger = {};
+const logMethods = ['log', 'debug', 'info', 'warn', 'error'];
 
-const devProxy = require('../DevProxy');
+beforeAll(() => {
+    logMethods.forEach(method => {
+        logger[method] = jest.fn();
+        jest.spyOn(console, method).mockImplementation();
+    });
+    devProxy = require('../DevProxy');
```

Comment:
What's the reason for requiring this inside a `beforeAll`? If we only need to `require` it once it should be fine in the top of the file

Reply:
I should leave a comment about this, but `http-proxy-middleware` destructures the console object at module definition time, so it evades Jest spies. By defining the spies before requiring DevProxy, we can actually spy on the proxy middleware's logging.

Reply:
Yeah, comment would be good because I would never assume that otherwise

Reply:
It seems like a pretty common thing about Jest mocking, unless I'm doing something consistently wrong. Which I've done before, with Jest!